src/main/java/com/denizenscript/clientizen/objects/properties/PropertyRegistry.java

```diff
@@ -0,0 +1,95 @@
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+import com.denizenscript.denizencore.utilities.debugging.DebugInternals;
+import net.minecraft.state.property.*;
+import net.minecraft.util.StringIdentifiable;
+
+import java.lang.reflect.Field;
+
+public class PropertyRegistry {
+
+    public static void register() {
+        registerBooleanProperty("switched", Properties.EYE, Properties.POWERED, Properties.ENABLED);
+        registerBooleanProperty("waterlogged", Properties.WATERLOGGED);
+        registerBooleanProperty("hanging", Properties.HANGING);
+        registerEnumProperty("bed_part", Properties.BED_PART);
+        registerEnumProperty("instrument", Properties.INSTRUMENT);
+        registerIntProperty("level", MaterialLevel.class);
+    }
+
+    public static void registerBooleanProperty(String name, BooleanProperty... properties) {
+        registerPropertyGetter(name, MaterialBooleanProperty::new, properties, MaterialBooleanProperty.class);
+    }
+
+    public static void registerIntProperty(String name, IntProperty... properties) {
+        registerPropertyGetter(name, MaterialIntProperty::new, properties, MaterialIntProperty.class);
+    }
+
+    @SafeVarargs
+    public static <T extends Enum<T> & StringIdentifiable> void registerEnumProperty(String name, EnumProperty<T>... properties) {
+        registerPropertyGetter(name, MaterialEnumProperty::new, properties, MaterialEnumProperty.class);
+    }
+
+    public static void registerBooleanProperty(String name, Class<? extends MaterialBooleanProperty> propertyClass) {
+        registerPropertyGetter(name, MaterialBooleanProperty::new, getHandledProperties(propertyClass), propertyClass);
+    }
+
+    public static void registerIntProperty(String name, Class<? extends MaterialIntProperty> propertyClass) {
+        registerPropertyGetter(name, MaterialIntProperty::new, getHandledProperties(propertyClass), propertyClass);
+    }
+
+    public static <T extends Enum<T> & StringIdentifiable, P extends MaterialEnumProperty<T>> void registerEnumProperty(String name, Class<P> propertyClass) {
+        registerPropertyGetter(name, MaterialEnumProperty::new, getHandledProperties(propertyClass), propertyClass);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T extends Property<?>> T[] getHandledProperties(Class<? extends MaterialMinecraftProperty<T, ?>> propertyClass) {
+        try {
+            Field handledField = propertyClass.getDeclaredField("handledProperties");
+            return (T[]) handledField.get(null);
+        }
+        catch (IllegalAccessException | NoSuchFieldException e) {
+            Debug.echoError("Invalid handled properties array for property class '" + DebugInternals.getClassNameOpti(propertyClass) + "': " + e.getMessage() + "!");
+        }
+        return null;
+    }
```

Comment:
I think you should remove all of these methods. Reflection is not needed here. Is there a way you can just pass in the required elements from the custom property class into `registerPropertyGetter`? For example, `MaterialLevel`'s register call would be `registerPropertyGetter("level", MaterialIntProperty::new, MaterialLevel.handledProperties, MaterialLevel.class);` Trying to simplify this with these methods yields diminishing returns, imo

Reply:
Fair point, it is a little more verbose but I suppose it does clean it up